packages/mwp-core/src/renderers/server-render.jsx

```diff
@@ -233,80 +233,87 @@ const makeRenderer = (
-		config: {
-			apiUrl: API_ROUTE_PATH,
-			baseUrl: host,
-			enableServiceWorker,
-			requestLanguage,
-			supportedLangs,
-			initialNow: new Date().getTime(),
-			variants: getVariants(state),
-			entryPath: url.pathname, // the path that the user entered the app on
-			media: getMedia(userAgent, userAgentDevice),
-		},
-	};
+	const initializeStore = () =>
+		request.server['mwp-app-route-plugin'].getFlags(memberId).then(flags => {
+			const initialState = {
+				config: {
+					apiUrl: API_ROUTE_PATH,
+					baseUrl: host,
+					enableServiceWorker,
+					requestLanguage,
+					supportedLangs,
+					initialNow: new Date().getTime(),
+					variants: getVariants(state),
+					entryPath: url.pathname, // the path that the user entered the app on
+					media: getMedia(userAgent, userAgentDevice),
+				},
+				flags,
+			};
 
-	const createStore = getServerCreateStore(
-		getRouteResolver(routes, basename),
-		middleware,
-		request
-	);
-	const store = createStore(reducer, initialState);
+			const createStore = getServerCreateStore(
+				getRouteResolver(routes, basename),
+				middleware,
+				request
+			);
+			return Promise.resolve(createStore(reducer, initialState));
+		});
 
 	// render skeleton if requested - the store is ready
-	if ('skeleton' in request.query) {
-		return Promise.resolve({
-			result: getHtml(
-				<Dom
-					basename={basename}
-					head={Helmet.rewind()}
-					initialState={store.getState()}
-					scripts={scripts}
-					cssLinks={cssLinks}
-					inlineStyleTags={getInlineStyleTags()}
-				/>
-			),
-			statusCode: 200,
-		});
-	}
 
 	// otherwise render using the API and React router
 	const checkReady = state =>
 		state.preRenderChecklist.every(isReady => isReady);
+	const populateStore = store =>
+		new Promise((resolve, reject) => {
+			// dispatch SERVER_RENDER to kick off API middleware
+			store.dispatch({ type: SERVER_RENDER, payload: url });
 
-	const initializeStore = new Promise((resolve, reject) => {
-		// dispatch SERVER_RENDER to kick off API middleware
-		store.dispatch({ type: SERVER_RENDER, payload: url });
-
-		if (checkReady(store.getState())) {
-			resolve(store);
-			return;
-		}
-		const unsubscribe = store.subscribe(() => {
 			if (checkReady(store.getState())) {
 				resolve(store);
-				unsubscribe();
+				return;
 			}
+			const unsubscribe = store.subscribe(() => {
+				if (checkReady(store.getState())) {
+					resolve(store);
+					unsubscribe();
+				}
+			});
 		});
-	});
 
-	return initializeStore.then(
-		initializedStore =>
-			// create tracer and immediately invoke the resulting function.
-			// trace should start before rendering, finish after rendering
-			newrelic.createTracer('serverRender', getRouterRenderer)({
-				routes,
-				store: initializedStore,
-				location: url,
-				basename,
-				scripts,
-				cssLinks,
-				userAgent,
-			}) // immediately invoke callback
-	);
+	return initializeStore().then(store => {
+		if ('skeleton' in request.query) {
```

Comment:
where does 'skeleton' come from?

Reply:
this is an old part of the offline-first work we did way back when. It's still useful, in theory, but is existing code that isn't really affected by this ticket

Reply:
cool cool